packages/core/utils/src/modules-sdk/define-link.ts

```diff
@@ -375,3 +427,74 @@ ${serviceBObj.module}: {
 
   return output
 }
+
+/**
+ * Generate a ModuleJoinerConfig for the read-only link definition on the fly.
+ *
+ * @param leftService
+ * @param rightService
+ * @param readOnlyLinkOptions
+ */
+export function defineReadOnlyLink(
+  leftService: DefineReadOnlyLinkInputSource,
+  property: string,
+  foreignKey: string,
+  rightService: DefineReadOnlyLinkInputSource,
+  readOnlyLinkOptions?: ReadOnlyOptions
+): void {
```

Comment:
thought: 

I have two suggestions this this:
- We should consider passing an object to make the purpose of each argument more explicit
- We should try to get as close to the `defineLink` API as possible

Perhaps, something like the following:
```
defineReadOnlyLink(
  {
    linkable: CustomModule.linkable.customModel,
    field: "variant_id"
  },
  ProductModule.linkable.productVariant,
)
```
```
defineLink(
  {
    linkable: CustomModule.linkable.customModel,
    field: "variant_id"
  },
  linkable: ProductModule.linkable.variant,
  {
    readOnly: true,
  }
)
```

Or something like that. We can discuss a more specific API, but just wanted to hear your thoughts on this first.

Reply:
I think we can use `defineLink` and the option `readOnly: true`.
Note that "linkable" might not exist in 1 or both sides, so the objects can be passed and what will be required is that the first argument requires `field`, and the second `alias`.

something like this:

```typescript
defineLink(
  {
    serviceName: "CustomModule",
    field: "variant_id",
  },
  ProductModule.linkable.productVariant.id,
  {
    readOnly: true,
  }
);

defineLink(
  {
    serviceName: "CustomModule",
    field: "variant_id",
  },
  {
    ...ProductModule.linkable.productVariant.id,
    alias: "my_variant",
  },
  {
    readOnly: true,
  }
);
```

Reply:
How do you know what data model from `CustomModule` the read only link applies to here/

```
defineLink(
  {
    serviceName: "CustomModule",
    field: "variant_id",
  }
  ...
```

Reply:
when no entity is provided, it will always default to the method `.list` from the module. If one specific data model is the target, the `entity` as to be set with the `serviceName`. Using the DML and `linkable` this is transparent, but Custom modules can be anything, not only a module created with DML.

Reply:
> This example is supported.

Great, then this LGTM!